File: openapi/typescript-fetch/models/V3BulkDeparturesUpdateResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * PTV Timetable API - Version 3
 * The PTV Timetable API provides direct access to Public Transport Victoria\'s public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
 *
 * The version of the OpenAPI document: v3
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
import type { V3BulkDeparturesRouteDirectionResponse } from './V3BulkDeparturesRouteDirectionResponse';
import {
    V3BulkDeparturesRouteDirectionResponseFromJSON,
    V3BulkDeparturesRouteDirectionResponseFromJSONTyped,
    V3BulkDeparturesRouteDirectionResponseToJSON,
} from './V3BulkDeparturesRouteDirectionResponse';
import type { V3Departure } from './V3Departure';
import {
    V3DepartureFromJSON,
    V3DepartureFromJSONTyped,
    V3DepartureToJSON,
} from './V3Departure';

/**
 * 
 * @export
 * @interface V3BulkDeparturesUpdateResponse
 */
export interface V3BulkDeparturesUpdateResponse {
    /**
     * Timetabled and real-time service departures
     * @type {Array<V3Departure>}
     * @memberof V3BulkDeparturesUpdateResponse
     */
    departures?: Array<V3Departure>;
    /**
     * Transport mode identifier
     * @type {number}
     * @memberof V3BulkDeparturesUpdateResponse
     */
    routeType?: number;
    /**
     * Stop identifier
     * @type {number}
     * @memberof V3BulkDeparturesUpdateResponse
     */
    stopId?: number;
    /**
     * 
     * @type {V3BulkDeparturesRouteDirectionResponse}
     * @memberof V3BulkDeparturesUpdateResponse
     */
    requestedRouteDirection?: V3BulkDeparturesRouteDirectionResponse;
    /**
     * The status of the route direction (changed | unchanged).
     * If changed, requests should change the requested_route_direction for the route_direction supplied.
     * @type {number}
     * @memberof V3BulkDeparturesUpdateResponse
     */
    routeDirectionStatus?: V3BulkDeparturesUpdateResponseRouteDirectionStatusEnum;
    /**
     * 
     * @type {V3BulkDeparturesRouteDirectionResponse}
     * @memberof V3BulkDeparturesUpdateResponse
     */
    routeDirection?: V3BulkDeparturesRouteDirectionResponse;
}


/**
 * @export
 */
export const V3BulkDeparturesUpdateResponseRouteDirectionStatusEnum = {
    NUMBER_0: 0,
    NUMBER_1: 1
} as const;
export type V3BulkDeparturesUpdateResponseRouteDirectionStatusEnum = typeof V3BulkDeparturesUpdateResponseRouteDirectionStatusEnum[keyof typeof V3BulkDeparturesUpdateResponseRouteDirectionStatusEnum];


/**
 * Check if a given object implements the V3BulkDeparturesUpdateResponse interface.
 */
export function instanceOfV3BulkDeparturesUpdateResponse(value: object): boolean {
    return true;
}

export function V3BulkDeparturesUpdateResponseFromJSON(json: any): V3BulkDeparturesUpdateResponse {
    return V3BulkDeparturesUpdateResponseFromJSONTyped(json, false);
}

export function V3BulkDeparturesUpdateResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): V3BulkDeparturesUpdateResponse {
    if (json == null) {
        return json;
    }
    return {
        
        'departures': json['departures'] == null ? undefined : ((json['departures'] as Array<any>).map(V3DepartureFromJSON)),
        'routeType': json['route_type'] == null ? undefined : json['route_type'],
        'stopId': json['stop_id'] == null ? undefined : json['stop_id'],
        'requestedRouteDirection': json['requested_route_direction'] == null ? undefined : V3BulkDeparturesRouteDirectionResponseFromJSON(json['requested_route_direction']),
        'routeDirectionStatus': json['route_direction_status'] == null ? undefined : json['route_direction_status'],
        'routeDirection': json['route_direction'] == null ? undefined : V3BulkDeparturesRouteDirectionResponseFromJSON(json['route_direction']),
    };
}

export function V3BulkDeparturesUpdateResponseToJSON(value?: V3BulkDeparturesUpdateResponse | null): any {
    if (value == null) {
        return value;
    }
    return {
        
        'departures': value['departures'] == null ? undefined : ((value['departures'] as Array<any>).map(V3DepartureToJSON)),
        'route_type': value['routeType'],
        'stop_id': value['stopId'],
        'requested_route_direction': V3BulkDeparturesRouteDirectionResponseToJSON(value['requestedRouteDirection']),
        'route_direction_status': value['routeDirectionStatus'],
        'route_direction': V3BulkDeparturesRouteDirectionResponseToJSON(value['routeDirection']),
    };
}

File: swagger.2.0/swagger-2/java/src/main/java/io/swagger/client/model/V3BulkDeparturesUpdateResponse.java
/*
 * PTV Timetable API - Version 3
 * The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
 *
 * OpenAPI spec version: v3
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import io.swagger.client.model.V3BulkDeparturesRouteDirectionResponse;
import io.swagger.client.model.V3Departure;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

/**
 * V3BulkDeparturesUpdateResponse
 */
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2024-04-22T22:15:30.079+10:00")
public class V3BulkDeparturesUpdateResponse {
  @SerializedName("departures")
  private List<V3Departure> departures = null;

  @SerializedName("route_type")
  private Integer routeType = null;

  @SerializedName("stop_id")
  private Integer stopId = null;

  @SerializedName("requested_route_direction")
  private V3BulkDeparturesRouteDirectionResponse requestedRouteDirection = null;

  /**
   * The status of the route direction (changed | unchanged).  If changed, requests should change the requested_route_direction for the route_direction supplied.
   */
  @JsonAdapter(RouteDirectionStatusEnum.Adapter.class)
  public enum RouteDirectionStatusEnum {
    NUMBER_0(0),
    
    NUMBER_1(1);

    private Integer value;

    RouteDirectionStatusEnum(Integer value) {
      this.value = value;
    }

    public Integer getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    public static RouteDirectionStatusEnum fromValue(String text) {
      for (RouteDirectionStatusEnum b : RouteDirectionStatusEnum.values()) {
        if (String.valueOf(b.value).equals(text)) {
          return b;
        }
      }
      return null;
    }

    public static class Adapter extends TypeAdapter<RouteDirectionStatusEnum> {
      @Override
      public void write(final JsonWriter jsonWriter, final RouteDirectionStatusEnum enumeration) throws IOException {
        jsonWriter.value(enumeration.getValue());
      }

      @Override
      public RouteDirectionStatusEnum read(final JsonReader jsonReader) throws IOException {
        int value = jsonReader.nextInt();
        return RouteDirectionStatusEnum.fromValue(String.valueOf(value));
      }
    }
  }

  @SerializedName("route_direction_status")
  private RouteDirectionStatusEnum routeDirectionStatus = null;

  @SerializedName("route_direction")
  private V3BulkDeparturesRouteDirectionResponse routeDirection = null;

  public V3BulkDeparturesUpdateResponse departures(List<V3Departure> departures) {
    this.departures = departures;
    return this;
  }

  public V3BulkDeparturesUpdateResponse addDeparturesItem(V3Departure departuresItem) {
    if (this.departures == null) {
      this.departures = new ArrayList<V3Departure>();
    }
    this.departures.add(departuresItem);
    return this;
  }

   /**
   * Timetabled and real-time service departures
   * @return departures
  **/
  @ApiModelProperty(value = "Timetabled and real-time service departures")
  public List<V3Departure> getDepartures() {
    return departures;
  }

  public void setDepartures(List<V3Departure> departures) {
    this.departures = departures;
  }

  public V3BulkDeparturesUpdateResponse routeType(Integer routeType) {
    this.routeType = routeType;
    return this;
  }

   /**
   * Transport mode identifier
   * @return routeType
  **/
  @ApiModelProperty(value = "Transport mode identifier")
  public Integer getRouteType() {
    return routeType;
  }

  public void setRouteType(Integer routeType) {
    this.routeType = routeType;
  }

  public V3BulkDeparturesUpdateResponse stopId(Integer stopId) {
    this.stopId = stopId;
    return this;
  }

   /**
   * Stop identifier
   * @return stopId
  **/
  @ApiModelProperty(value = "Stop identifier")
  public Integer getStopId() {
    return stopId;
  }

  public void setStopId(Integer stopId) {
    this.stopId = stopId;
  }

  public V3BulkDeparturesUpdateResponse requestedRouteDirection(V3BulkDeparturesRouteDirectionResponse requestedRouteDirection) {
    this.requestedRouteDirection = requestedRouteDirection;
    return this;
  }

   /**
   * The route direction that these departures are for. Will be one of the requested route directions
   * @return requestedRouteDirection
  **/
  @ApiModelProperty(value = "The route direction that these departures are for. Will be one of the requested route directions")
  public V3BulkDeparturesRouteDirectionResponse getRequestedRouteDirection() {
    return requestedRouteDirection;
  }

  public void setRequestedRouteDirection(V3BulkDeparturesRouteDirectionResponse requestedRouteDirection) {
    this.requestedRouteDirection = requestedRouteDirection;
  }

  public V3BulkDeparturesUpdateResponse routeDirectionStatus(RouteDirectionStatusEnum routeDirectionStatus) {
    this.routeDirectionStatus = routeDirectionStatus;
    return this;
  }

   /**
   * The status of the route direction (changed | unchanged).  If changed, requests should change the requested_route_direction for the route_direction supplied.
   * @return routeDirectionStatus
  **/
  @ApiModelProperty(value = "The status of the route direction (changed | unchanged).  If changed, requests should change the requested_route_direction for the route_direction supplied.")
  public RouteDirectionStatusEnum getRouteDirectionStatus() {
    return routeDirectionStatus;
  }

  public void setRouteDirectionStatus(RouteDirectionStatusEnum routeDirectionStatus) {
    this.routeDirectionStatus = routeDirectionStatus;
  }

  public V3BulkDeparturesUpdateResponse routeDirection(V3BulkDeparturesRouteDirectionResponse routeDirection) {
    this.routeDirection = routeDirection;
    return this;
  }

   /**
   * The route direction found matching the requested_route_direction
   * @return routeDirection
  **/
  @ApiModelProperty(value = "The route direction found matching the requested_route_direction")
  public V3BulkDeparturesRouteDirectionResponse getRouteDirection() {
    return routeDirection;
  }

  public void setRouteDirection(V3BulkDeparturesRouteDirectionResponse routeDirection) {
    this.routeDirection = routeDirection;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3BulkDeparturesUpdateResponse v3BulkDeparturesUpdateResponse = (V3BulkDeparturesUpdateResponse) o;
    return Objects.equals(this.departures, v3BulkDeparturesUpdateResponse.departures) &&
        Objects.equals(this.routeType, v3BulkDeparturesUpdateResponse.routeType) &&
        Objects.equals(this.stopId, v3BulkDeparturesUpdateResponse.stopId) &&
        Objects.equals(this.requestedRouteDirection, v3BulkDeparturesUpdateResponse.requestedRouteDirection) &&
        Objects.equals(this.routeDirectionStatus, v3BulkDeparturesUpdateResponse.routeDirectionStatus) &&
        Objects.equals(this.routeDirection, v3BulkDeparturesUpdateResponse.routeDirection);
  }

  @Override
  public int hashCode() {
    return Objects.hash(departures, routeType, stopId, requestedRouteDirection, routeDirectionStatus, routeDirection);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3BulkDeparturesUpdateResponse {\n");
    
    sb.append("    departures: ").append(toIndentedString(departures)).append("\n");
    sb.append("    routeType: ").append(toIndentedString(routeType)).append("\n");
    sb.append("    stopId: ").append(toIndentedString(stopId)).append("\n");
    sb.append("    requestedRouteDirection: ").append(toIndentedString(requestedRouteDirection)).append("\n");
    sb.append("    routeDirectionStatus: ").append(toIndentedString(routeDirectionStatus)).append("\n");
    sb.append("    routeDirection: ").append(toIndentedString(routeDirection)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

